emloop/hooks/abstract_hook.py

```diff
@@ -102,3 +104,14 @@ def after_training(self) -> None:
             This method is called exactly once during the training.
         """
         pass
+
+    def register_mainloop(self, main_loop: 'emloop.MainLoop') -> None:
+        """
+        Pass MainLoop to hook. Raise ValueError if MainLoop was already passed before.
```

Comment:
use correct rst formatting for `ValueError` and especially `MainLoop` so that the generated documentation is clickable